Positron Notebooks UI sprint

nstrayer opened the pull request:
Addresses #9636.

This PR implements several UI polish fixes for the new notebooks frontend from the design sprint:



### Error state indication (bcbfbfea7242fe75028b61d2048559ebcc7a55df)

When a cell execution fails, the selection bar and execution badge now display in red using the error foreground color 

<img width="462" height="320" alt="image" src="https://github.com/user-attachments/assets/fb670491-a7db-42ee-87b7-e20b517c0b60" />



### Gutter spacing improvements (dcadd3cd76a1ce6594c08ddf296a71c4d322b84e)

Fixed overlapping between gutter elements (like breakpoints) and the selection bar by moving the entire gutter system over and using the selection bar as a background extension 


> [!WARNING]
> This may conflict with #9860 but does it in a different way that should make it easier to handle the change to selection + active cells when it's added for #9848. I am totally fine dropping the commit for this though to avoid conflicts/confusion.


<img width="214" height="238" alt="image" src="https://github.com/user-attachments/assets/bfa6c708-69c6-485c-a2d5-84fe42ef2963" />



### Terminology update (f6a777fe80db0fcbd0834ad0d3811386acab44af)


Changed all instances of "markup" to "markdown" for consistency with VS Code language 

<img width="412" height="363" alt="image" src="https://github.com/user-attachments/assets/df3a4b80-da1f-4849-9a45-a146eb2dd817" />


### Cell action menu padding (1fd9e6be56f29315885ce8d182aa169c49ab8a60)

Fixed padding issues in the NotebookCellMoreActionsMenu component by refactoring to use the ActionButton component 


<img width="342" height="137" alt="image" src="https://github.com/user-attachments/assets/86703d52-e8a8-4ce7-9ec0-a0047d4e001e" />

> [!WARNING]
> Again may conflict a bit with #9860 but gets at the underlying problem of the wrong button component being used for the more actions expansion.



---

### Release Notes

#### New Features
- N/A

#### Bug Fixes

### QA Notes

@:positron-notebooks

**Testing error indication:**
1. Open a notebook
2. Create a code cell with intentionally failing code (e.g., `raise Exception("test error")` for Python)
3. Execute the cell
4. Verify that when the cell fails:
   - The left selection bar turns red
   - The execution badge `[N]` displays in red
   - The editor outline (when focused) uses the error color

**Testing selection bar spacing:**
1. Open a notebook
2. Set a breakpoint in a code cell
3. Select the cell
4. Verify the breakpoint indicator doesn't overlap with the left selection bar
5. Verify proper spacing between the selection bar and editor content

**Testing markdown terminology:**
1. Open a notebook
2. Add a markdown cell
3. Verify all UI text refers to "markdown" rather than "markup"
4. Check tooltip text and empty cell messages

**Testing cell action menu:**
1. Hover over a cell to reveal the action bar
2. Click the ellipsis (...) button to open the more actions menu
3. Verify proper padding around menu items

nstrayer committed:
Add CSS styles for error indication in execution badges and selection bars.

Fix outline colors for error cells

nstrayer committed:
Change phrase from "markup" -> "markdown"

nstrayer committed:
Fix gutter elements overlapping with selection bar by moving whole system over a bit and using selection bar act as a background extension when the cell is not selected.

nstrayer committed:
Fix padding on the cell action menu

seeM commented on src/vs/workbench/contrib/positronNotebook/browser/notebookCells/CellLeftActionMenu.tsx:
```diff
@@ -51,6 +51,7 @@ export function CellLeftActionMenu({ cell }: CellLeftActionMenuProps) {
 	const executionStatus = useObservedValue(cell.executionStatus);
 	const duration = useObservedValue(cell.lastExecutionDuration);
 	const lastRunEndTime = useObservedValue(cell.lastRunEndTime);
+	const hasError = useObservedValue(cell.hasError);
```
Might be better to avoid tracking this state both here and in `NotebookCellWrapper` and making one a prop? We might be doing that in a couple places

seeM reviewed (dismissed):
Code changes look great! UI changes look good as well but I'll leave for others to review in more detail

I think your UI solutions are better than mine from https://github.com/posit-dev/positron/pull/9860!

cindyytong commented:
LGTM thank you

nstrayer committed:
Refactor hasError to use derived observable pattern

Addresses PR #9998 review feedback to decouple hasError from outputs
observable and eliminate duplicate state tracking. Makes hasError an
abstract property implemented by each cell type: code cells derive it
from outputs, markdown cells use a constant false value.

nstrayer committed:
Fix dropdown menu by passing ref correctly in notebook action button

nstrayer committed:
Restructure hasError to be a derived and passed down prop.
